pkg/dashboard/api_test.go

```diff
@@ -0,0 +1,46 @@
+package dashboard
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/gin-gonic/gin"
+	"github.com/komodorio/helm-dashboard/pkg/dashboard/handlers"
+	"github.com/komodorio/helm-dashboard/pkg/dashboard/objects"
+	"gotest.tools/v3/assert"
+)
+
+func GetTestGinContext(w *httptest.ResponseRecorder) *gin.Context {
+	gin.SetMode(gin.TestMode)
+
+	ctx, _ := gin.CreateTestContext(w)
+	ctx.Request = &http.Request{
+		Header: make(http.Header),
+	}
+
+	return ctx
+}
+
+func TestNoCacheMiddleware(t *testing.T) {
+	w := httptest.NewRecorder()
+	con := GetTestGinContext(w)
+	noCache(con)
+	assert.Equal(t, w.Header().Get("Cache-Control"), "no-cache")
+}
+
+func TestEnableCacheControl(t *testing.T) {
+	w := httptest.NewRecorder()
+	con := GetTestGinContext(w)
+
+	// Sets deafault policy to `no-cache`
+	noCache(con)
```

Comment:
The sequence here is wrong . noCache should be called after EnableClientCache, that's how it happens in the code we want to cover. Just move this line after EnableClientCache and you're done.

Reply:
But that makes the test to fail
Since `no-cache` is middleware `no-cache` gets called first right ?

Reply:
Oh, ok 
Maybe you're right